pkg/apis/pingcap/v1alpha1/types.go

```diff
@@ -104,62 +104,91 @@ type TidbClusterList struct {
-	EnablePVReclaim bool `json:"enablePVReclaim,omitempty"`
+	// ImagePullPolicy of TiDB cluster Pods
+	// +kubebuilder:default=IfNotPresent
+	ImagePullPolicy corev1.PullPolicy `json:"imagePullPolicy,omitempty"`
 
-	// Enable TLS connection between TiDB server components
-	EnableTLSCluster bool `json:"enableTLSCluster,omitempty"`
+	// ConfigUpdateStrategy determines how the configuration change is applied to the cluster.
+	// UpdateStrategyInPlace will update the ConfigMap of configuration in-place and an extra rolling-update of the
+	// cluster component is needed to reload the configuration change.
+	// UpdateStrategyRollingUpdate will create a new ConfigMap with the new configuration and rolling-update the the
+	// related components to use the new ConfigMap, that is, the new configuration will be applied automatically.
+	// +kubebuilder:validation:Enum=InPlace,RollingUpdate
+	// +kubebuilder:default=InPlacne
+	ConfigUpdateStrategy ConfigUpdateStrategy `json:"configUpdateStrategy,omitempty"`
 
-	// Time zone of TiDB cluster Pods
-	Timezone string `json:"timezone,omitempty"`
+	// Whether enable PVC reclaim for orphan PVC left by statefulset scale-in
+	// Optional: Defaults to false
+	// +optional
+	EnablePVReclaim *bool `json:"enablePVReclaim,omitempty"`
 
-	// ImagePullPolicy of TiDB cluster Pods
-	ImagePullPolicy corev1.PullPolicy `json:"imagePullPolicy,omitempty"`
+	// Enable TLS connection between TiDB server components
+	// Optional: Defaults to false
+	// +optional
+	EnableTLSCluster *bool `json:"enableTLSCluster,omitempty"`
 
 	// Whether Hostnetwork is enabled for TiDB cluster Pods
-	HostNetwork bool `json:"hostNetwork,omitempty"`
+	// Optional: Defaults to false
+	// +optional
+	HostNetwork *bool `json:"hostNetwork,omitempty"`
 
 	// Affinity of TiDB cluster Pods
+	// +optional
 	Affinity *corev1.Affinity `json:"affinity,omitempty"`
 
 	// PriorityClassName of TiDB cluster Pods
-	PriorityClassName string `json:"priorityClassName,omitempty"`
+	// Optional: Defaults to omitted
+	// +optional
+	PriorityClassName *string `json:"priorityClassName,omitempty"`
 
 	// Base node selectors of TiDB cluster Pods, components may add or override selectors upon this respectively
+	// +optional
 	NodeSelector map[string]string `json:"nodeSelector,omitempty"`
 
 	// Base annotations of TiDB cluster Pods, components may add or override selectors upon this respectively
+	// +optional
 	Annotations map[string]string `json:"annotations,omitempty"`
 
 	// Base tolerations of TiDB cluster Pods, components may add more tolreations upon this respectively
+	// +optional
 	Tolerations []corev1.Toleration `json:"tolerations,omitempty"`
+
+	// Time zone of TiDB cluster Pods
+	// Optional: Defaults to UTC
+	// +optional
+	Timezone string `json:"timezone,omitempty"`
+
+	// Services list non-headless services type used in TidbCluster
+	// Deprecated
+	// +k8s:openapi-gen=false
+	Services []Service `json:"services,omitempty"`
```

Comment:
For deprecated fields, skip the openapi gen, so that they could be hidden from user in the schema and documents.